src/net/l0/net_entry.c

```diff
@@ -30,75 +32,114 @@ static DLIST_DEFINE(netif_rx_list);
 
 static struct lthread netif_rx_irq_handler;
 
-static void netif_rx_queued(struct net_device *dev) {
-	ipl_t sp;
-
-	assert(dev != NULL);
+static int netif_rx_action(struct lthread *self) {
+	struct net_device *dev;
+	ipl_t ipl;
 
-	sp = ipl_save();
+	ipl= ipl_save();
 	{
-		if (dlist_empty(&dev->rx_lnk)) {
-			dlist_add_prev(&dev->rx_lnk, &netif_rx_list);
+		dlist_foreach_entry_safe(dev, &netif_rx_list, rx_lnk) {
+			struct sk_buff *skb;
+
+			while ((skb = skb_queue_pop(&dev->dev_queue)) != NULL) {
+				ipl_restore(ipl);
+				{
+					net_rx(skb);
+				}
+				ipl= ipl_save();
+			}
+			dlist_del_init(&dev->rx_lnk);
 		}
 	}
-	ipl_restore(sp);
+	ipl_restore(ipl);
+
+	return 0;
 }
 
-static void netif_rx_dequeued(struct net_device *dev) {
-	ipl_t sp;
+/* we can be in irq mode */
+int netif_rx(void *data) {
```

Comment:
May be change this function signature to ` int netif_rx(struct sk_buff *skb) {` to make sure correct type is passed here?

Reply:
Thanks, 
It's the only for pnet support, but if it's changed pnet will breake